ocs_ci/ocs/scale_lib.py

```diff
@@ -998,3 +1000,251 @@ def get_max_pvc_count():
+
+    Returns:
+        pod_running_list (list): List of all PODs reached running state.
+
+    Asserts:
+        If not all POD reached Running state.
+
+    """
+
+    # Check all the POD reached Running state
+    pod_running_list, pod_not_running_list = ([] for i in range(2))
+    while_iteration_count = 0
+    while True:
+        # Get kube_job obj and fetch either all PODs are in Running state
+        # If not Running, adding those PODs to pod_not_running_list
+        job_get_output = kube_job_obj.get(namespace=namespace)
+        for i in range(no_of_pod):
+            if job_get_output["items"][0]["kind"] == constants.POD:
+                pod_type = constants.POD
+            else:
+                pod_type = None
+            if pod_type:
+                status = job_get_output["items"][i]["status"]["phase"]
+                logging.info(
+                    f"POD {job_get_output['items'][i]['metadata']['name']} status {status}"
+                )
+                if status != "Running":
+                    pod_not_running_list.append(
+                        job_get_output["items"][i]["metadata"]["name"]
+                    )
+            else:
+                # For DC config there is no Running status so checking it based on
+                # availableReplicas, basically this will be 1 if pod is running and
+                # the value will be 0 in-case of pod not in running state
+                status = job_get_output["items"][i]["status"]["availableReplicas"]
+                logging.info(
+                    f"DC Config {job_get_output['items'][i]['metadata']['name']} "
+                    f"available running pods {status}"
+                )
+                if not status:
+                    pod_not_running_list.append(
+                        job_get_output["items"][i]["metadata"]["name"]
+                    )
+
+        # Check the length of pod_not_running_list to decide either all PODs reached
+        # Running state, If not then wait for 30secs and re-iterate while loop
+        if len(pod_not_running_list):
+            time.sleep(timeout)
+            while_iteration_count += 1
+            # Breaking while loop after 10 Iteration i.e. after 30*10 secs of wait_time
+            # And if PODs are still not in Running state then there will be assert.
+            if while_iteration_count >= 10:
+                assert logging.error(
+                    f" Listed PODs took more than 300secs to bound {pod_not_running_list}"
+                )
+                break
+            pod_not_running_list.clear()
+            continue
+        elif not len(pod_not_running_list):
+            for i in range(no_of_pod):
+                pod_running_list.append(job_get_output["items"][i]["metadata"]["name"])
+            logging.info("All PODs are in Running state")
+            break
+
+    return pod_running_list
+
+
+def attach_multiple_pvc_to_pod_dict(
```

Comment:
Sample yaml file generated by this function can be found in http://pastebin.test.redhat.com/954350